ui/js/controllers/main.js

```diff
@@ -22,7 +22,59 @@ treeherderApp.controller('MainCtrl', [
 
         $rootScope.getWindowTitle = function() {
             var ufc = $scope.getUnclassifiedFailureCount($rootScope.repoName);
-            return "[" + ufc + "] " + $rootScope.repoName;
+            var params = $location.search();
+
+            // repoName is undefined for the first few title update attempts, show something sensible
+            var title = "[" + ufc + "] " + ($rootScope.repoName ? $rootScope.repoName : "Treeherder");
+
+            if (params["revision"]) {
+                var desc = getSingleRevisionTitleString();
+                var revtitle = desc[0] ? ": " + desc[0] : "";
```

Comment:
`getSingleRevisionTitleString()` can return `false`, in which case `desc[0]` is undefined.